libs/pika/mpi_base/src/mpi_environment.cpp

```diff
@@ -19,17 +20,16 @@
 #include <string>
 
 ///////////////////////////////////////////////////////////////////////////////
-namespace pika { namespace util {
+namespace pika { namespace mpi {
 
     namespace detail {
 
-        bool detect_mpi_environment(util::runtime_configuration const& cfg, char const* default_env)
+        bool detect_environment(util::runtime_configuration const& cfg, char const* default_env)
         {
 #if defined(__bgq__)
-            // If running on BG/Q, we can safely assume to always run in an
-            // MPI environment
+            // If running on BG/Q, we can safely assume to always run in an MPI environment
             return true;
-#else
+#endif
```

Comment:
Isn't changing this from `#else` to `#endif` going to lead to warnings about multiple returns from the function, with the second one being unused? I think semantically the `#else` was more appropriate.